diff -uraN umockdev-0.15.3/src/libumockdev-preload.c omv-umockdev-0.15.3/src/libumockdev-preload.c
--- umockdev-0.15.3/src/libumockdev-preload.c	2020-12-27 10:33:24.273351200 +0100
+++ omv-umockdev-0.15.3/src/libumockdev-preload.c	2020-12-28 13:47:45.639278891 +0100
@@ -909,7 +909,7 @@
     assert(r);
     r = fd_map_get(&script_dev_format_map, fd_dev, &data);
     assert(r);
-    fmt = (enum script_record_format) data;
+    fmt = (enum script_record_format) (long) data;
 
     DBG(DBG_SCRIPT, "script_record_open: start recording fd %i on device %i:%i into %s (format %i)\n",
 	fd, major(fd_dev), minor(fd_dev), logname, fmt);
